yew-functional/src/lib.rs

```diff
@@ -173,204 +169,76 @@ where
     }
 }
 
-pub fn use_ref<T: 'static, InitialProvider>(initial_value: InitialProvider) -> Rc<RefCell<T>>
-where
-    InitialProvider: FnOnce() -> T,
-{
-    #[derive(Clone)]
-    struct UseRefState<T>(Rc<RefCell<T>>);
-    impl<T> Hook for UseRefState<T> {}
-
-    use_hook(
-        |state: &mut UseRefState<T>, hook_callback| {
-            // we need it to be a specific closure type, even if we never use it
-            let _ignored = || hook_callback(|_| false, false);
-            state.0.clone()
-        },
-        move || UseRefState(Rc::new(RefCell::new(initial_value()))),
-    )
+pub fn get_current_scope() -> Option<AnyScope> {
+    CURRENT_HOOK.with(|cell| cell.borrow().as_ref().map(|state| state.scope.clone()))
 }
 
-pub fn use_reducer<Action: 'static, Reducer, State: 'static>(
-    reducer: Reducer,
-    initial_state: State,
-) -> (Rc<State>, Rc<impl Fn(Action)>)
-where
-    Reducer: Fn(Rc<State>, Action) -> State + 'static,
-{
-    use_reducer_with_init(reducer, initial_state, |a| a)
-}
+use std::ops::DerefMut;
```

Comment:
what's this doing down here? Please clean up the imports.

Reply:
Did my best, and ran `cargo fix`. LMK if you see anything out of order.